lib/services/address/index.js

```diff
@@ -305,6 +303,14 @@ AddressService.prototype.updateMempoolIndex = function(tx, add, callback) {
       continue;
     }
 
+    var hashBufferHex = addressInfo.hashBuffer.toString('hex');
+
+    if (add) {
+      this.mempoolAddressIndex[hashBufferHex] = true;
```

Comment:
We may need to include the type of address hash here as well. We can also half the memory used for this by doing something similar to: https://github.com/bitpay/bitcore-node/blob/opt/multiple-address-history/lib/services/address/index.js#L348